tests/unit/model/native-access-arrays-test.js

```diff
@@ -0,0 +1,182 @@
+                },
+              ],
+            },
+          },
+        })
+      );
+
+      let chapters = model.get('chapters');
+
+      assert.equal(
+        chapters[0].get('name'),
+        'The Boy Who Lived',
+        `chapters's embedded records can resolve values`
+      );
+      chapters.push({ name: 'The Vanishing Glass' });
+
+      let chapter2 = chapters[1];
+      assert.equal(chapter2.constructor.isModel, true, 'new values can be resolved via push');
+      assert.equal(
+        get(chapter2, 'name'),
+        'The Vanishing Glass',
+        `new values can be resolved via push`
+      );
+      let spreadChapters = [...chapters, { name: 'plain object' }];
+      assert.equal(
+        spreadChapters[0].get('name'),
+        'The Boy Who Lived',
+        `can use the spread operator with managed arrays`
+      );
+      assert.equal(
+        chapters.some((el) => el.get('name') === 'The Boy Who Lived'),
+        true,
+        'array.some() is implemented'
+      );
+      chapters.sort((first, second) => (first.get('name') < second.get('name') ? 1 : -1));
+      assert.equal(chapters[0].get('name'), 'The Vanishing Glass', `Array.sort is implemented`);
+
+      let slicedChapters = chapters.slice(1);
+      assert.equal(slicedChapters[0].get('name'), 'The Boy Who Lived', `Array.slice still works`);
+      assert.notEqual(slicedChapters, chapters, `Array.slice still works`);
+
+      let shiftedChapters = chapters.slice();
+      let shiftedChapter = shiftedChapters.shift();
+
+      assert.equal(shiftedChapters[0].get('name'), 'The Boy Who Lived', `Array.shift works`);
+      assert.equal(shiftedChapters.length, 1, `Array.shift works`);
+      assert.equal(shiftedChapter.get('name'), 'The Vanishing Glass', 'Array.shift works');
+
+      shiftedChapters.unshift(shiftedChapter);
+      assert.equal(shiftedChapters[0].get('name'), 'The Vanishing Glass', `Array.unshift works`);
+      assert.equal(shiftedChapters.length, 2, `Array.unshift works`);
+
+      let toReverse = chapters.slice();
+      toReverse.reverse();
+      assert.equal(toReverse[0].get('name'), 'The Boy Who Lived', `Array.reverse works`);
+      assert.equal(toReverse[1].get('name'), 'The Vanishing Glass', `Array.reverse works`);
+
+      assert.equal(
+        chapters.reduce((acc, item) => acc + ' and ' + item.get('name'), ''),
+        ' and The Vanishing Glass and The Boy Who Lived',
+        'Array.reduce still works'
+      );
+      assert.equal(
+        chapters.reduceRight((acc, item) => acc + ' and ' + item.get('name'), ''),
+        ' and The Boy Who Lived and The Vanishing Glass',
+        'Array.reduceRight works'
+      );
+
+      assert.equal(
+        chapters.join(' '),
+        '<EmbeddedMegamorphicModel:undefined> <EmbeddedMegamorphicModel:undefined>',
+        `Array.join works`
+      );
+
+      if ([].at) {
+        assert.equal(chapters.at(-1).get('name'), 'The Boy Who Lived', `Array.at is implemented`);
+      }
```

Comment:
Does this mean this only runs sometimes?

Reply:
Only if the browser already supports `Array.prototype.at`. Chrome added it in latest version, I had to update Chrome before it would work.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at#browser_compatibility